mantidproject/mantid
# Check axis type before table transposition Refs #23155

**Description of work.**
This PR adds a check on the y-axis type (i.e. Text or Numeric) of the input workspace(s) and correctly extracts the axis values/labels into the parameters table and the result workspace.

**To test:**
- Open Indirect > Data Analysis > F(Q)
- Enter the provided test data
- Select 'Chudley Elliot' in the 'Fit Type' drop-down menu
- Click the 'Fit Single Spectrum Button'
- Ensure the fit completes successfully
- Open the *_Parameters workspace and ensure that the axis-1 column contains f1.f1.FWHM

**Test Files**
[IN16B_125878_QLd_Result.nxs.zip](https://github.com/mantidproject/mantid/files/2249669/IN16B_125878_QLd_Result.nxs.zip)

Fixes #23155 

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
